### PR TITLE
Move Traffic tab to PF4

### PR DESCRIPTION
**Pull request by aljesusg**
Fix https://github.com/kiali/kiali/issues/1808

## Apps
apps
## Workloads
workloads

## Services
service

### CODE REVIEW

**Comment by aljesusg** [comment]
Hi @jotak , I don't understand why this tests changed. Can you help me here? 

```bash
 FAIL  src/components/Metrics/__tests__/TrafficDetails.test.tsx (8.341s)
  ● TrafficDetails › pass down traffic - simple in-out graph two levels
    expect(received).toEqual(expected) // deep equality
    Expected: "svc1,->wk1"
    Received: "svc1,wk1"
      222 | 
      223 |     expect(inboundList).toHaveLength(2);
    > 224 |     expect(inboundList.join()).toEqual('svc1,->wk1');
          |                                ^
      225 |     expect(outboundList).toHaveLength(2);
      226 |     expect(outboundList.join()).toEqual('svc2,->wk2');
      227 |   });
      at Object.toEqual (src/components/Metrics/__tests__/TrafficDetails.test.tsx:224:32)
  ● TrafficDetails › pass down traffic - slightly more complex inbound
    expect(received).toEqual(expected) // deep equality
    Expected: "svc1,->wk1,->wk2,svc2,->wk2,wk3"
    Received: "svc1,wk1,wk2,svc2,wk2,wk3"
      247 | 
      248 |     expect(inboundList).toHaveLength(6);
    > 249 |     expect(inboundList.join()).toEqual('svc1,->wk1,->wk2,svc2,->wk2,wk3');
          |                                ^
      250 |   });
      251 | });
      252 | 
      at Object.toEqual (src/components/Metrics/__tests__/TrafficDetails.test.tsx:249:32)
```

**Comment by jotak** [comment]
@aljesusg Looks like the changes need some additional adaptation in the tests. As far as I understand, in _TrafficDetails.test.tsx_ line 113-123 it tries to find icon to destination(?) items and then return that `->` prefix to make the distinction with the source items ... or something like that, is it correct @israel-hdez  ?

So I guess it's the `item.find('Icon')` that needs to be updated here.

**Comment by jotak** [comment]
@aljesusg suggestion: replace

```javascript
const icon = item.find('Icon');
if (icon.length > 1 && icon.at(1).prop('style')!.paddingLeft) {
```

with:

```javascript
      const icon = item.find('BundleIcon');
      if (icon.exists() && icon.prop('style')!.paddingLeft) {
```

**Comment by jotak** [comment]
Beside that, looks like there are several broken things:

**Before**:

Capture d’écran de 2019-10-16 13-43-20

**After:**

Capture d’écran de 2019-10-16 13-43-23

- Health color is missing
- Workloads icon is not displayed
- Service (and presumably workload) icons look tiny

**Comment by aljesusg** [comment]
Fixed tests and colors @jotak , it seems that PF4 icons currentColor is gray by default

**Comment by israel-hdez** [comment]
> @aljesusg Looks like the changes need some additional adaptation in the tests. As far as I understand, in TrafficDetails.test.tsx line 113-123 it tries to find icon to destination(?) items and then return that -> prefix to make the distinction with the source items ... or something like that, is it correct @israel-hdez ?
So I guess it's the item.find('Icon') that needs to be updated here.

If I understood correctly, that's right.
The "->" prefix is added to the rows that has indentation in the source/destination column. Rows without indentation won't have the prefix. That's how I'm testing that rows have the right indentation (suggestions to improve that are welcome)

File: src/components/Details/__tests__/DetailedTrafficList.test.tsx [hunk 1]
```diff
@@ -159,47 +159,47 @@ describe('DetailedTrafficList', () => {
     const wrapper = shallow(<DetailedTrafficList direction={'outbound'} traffic={[trafficItem]} />);
 
     const cell = wrapper.find('TableGridCol').at(DetailedTrafficList.STATUS_COLUMN_IDX);
-    expect(cell.contains(<Icon type="pf" name="ok" />)).toBeTruthy();
+    expect(cell.contains(<OkIcon />)).toBeTruthy();
```
Comment (israel-hdez) [inline src/components/Details/__tests__/DetailedTrafficList.test.tsx]: I think you need to add the `color` attribute to have passing tests.

**Comment by aljesusg** [comment]
This could be the first step to remove PF3 :) 

- Removed patternfly-react-extensions
- Migrated Table
- Move tests

new

cc @hhovsepy @jotak

**Comment by aljesusg** [comment]
# Service Size
service
# Workload Size
workload

**Comment by israel-hdez** [comment]
Post-merge thumbs up :)
:+1:

**Pull request merged**